src/Microsoft.Health.Fhir.CosmosDb/Features/Search/Queries/QueryBuilder.cs

```diff
@@ -38,11 +38,11 @@ public QueryBuilderHelper()
                 _queryHelper = new QueryHelper(_queryBuilder, _queryParameterManager, SearchValueConstants.RootAliasName);
             }
 
-            public SqlQuerySpec BuildSqlQuerySpec(SearchOptions searchOptions)
+            public SqlQuerySpec BuildSqlQuerySpec(SearchOptions searchOptions, bool calculateTotalCount = false)
```

Comment:
I think we should change the signature to: `public SqlQuerySpec BuildSqlQuerySpec(Expression expression, bool countOnly)` or something like that. We should also update `GenerateHistorySql` below to match as well so `public SqlQuerySpec GenerateHistorySql(Expression expression)`. Actually, if you are touching it, might as well rename them so the names are consistent (e.g., GenerateSearchSql and GenerateHistorySql or something).

Reply:
As discussed, I created [#AB70828](https://microsofthealth.visualstudio.com/Health/_workitems/edit/70828).